src/audio/mixer.c

```diff
@@ -387,15 +382,9 @@ static int mixer_copy(struct comp_dev *dev)
 
 		buffer_release(sink_c);
 
-		md->sources_inactive = true;
-
 		return 0;
 	}
 
-	if (md->sources_inactive) {
-		md->sources_inactive = false;
-		comp_dbg(dev, "mixer_copy exit sources_inactive state");
-	}
```

Comment:
@kv2019i the mixer should mix the minimum amount of data available among all the active sources. But if an active source has no data available, it should be taken into account. Thats exactly wha this PR does. The idea of inactive_sources in rolled into the first for loop which checks if a source has any data available or not

Reply:
@kv2019i how would you tell if the source was just active or has been active for a while and producing samples previously or is coming out of a pause? The mixer would not introduce discontinuity on its own. It would introduce it if the source underruns and that's out of the mixer's jurisdiction right?

Reply:
@ranj063 With current pipeline definitions, this indeed requires mixer to maintain per-source state to make that call. A common cleaner solution is to use a dedicated component state (PRE_ROLLING) to model this state of a component, which can be observed by mixer (but also by pipeline framework code). I.e. source component won't become ACTIVE until it is ready to feed data at the downstream rate. This would be useful for deep buffering and also for processing components that process data at larger chunks.

If a single source is not keeping up and fails to deliver samples due to performance, I don't think mixer should try to cope with this. Rather it should starve the mixer output of data and if the situation persists, cause the whole pipeline to hit an XRUN. Some audio signal is anyways getting corrupted, so better to raise this as an error.

If we need to connect an unreliable source (let's say audio coming over network and streamed to DSP) to a mixer, I'd rather add a separate component between this source and the mixer, which is tasked to handle this. E.g. a jitter buffer component that has logic how to cope with missing segments of audio (e.g. with noise generation to mask the gap). This keeps the real-time mixer more simple in its logic.